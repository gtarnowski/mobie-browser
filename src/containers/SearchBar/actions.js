import { createAction } from 'redux-act';

export const searchBarValue = createAction('SEARCH_BAR_VALUE');
export const searchBarFilter = createAction('SEARCH_BAR_FILTER');
export const searchBarSubmitQuery = createAction('SEARCH_PAGE_SUBMIT_QUERY');
