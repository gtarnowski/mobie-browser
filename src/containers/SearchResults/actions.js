import { createAction } from 'redux-act';

export const searchResults = createAction('SEARCH_RESULTS');
export const searchResultsError = createAction('SEARCH_RESULTS_ERROR');
export const searchResultsLoading = createAction('SEARCH_RESULTS_LOADING');
