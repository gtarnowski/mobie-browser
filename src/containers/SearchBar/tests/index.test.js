import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar, mapDispatchToProps } from '../index';
import { searchBarSubmitQuery, searchBarFilter, searchBarValue } from '../actions';

describe('<SearchBar />', () => {
  describe('Render', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const onCheck = jest.fn();

    const defaultProps = {
      onChange,
      onSubmit,
      onCheck,
      searchValue: 'test',
      filter: 'all',
    };
    const createComponent = props => shallow(<SearchBar {...defaultProps} {...props} />);

    it('should render correctly', () => {
      const component = createComponent();
      expect(component).toMatchSnapshot();
    });

    it('should render correctly with inputs', () => {
      const component = createComponent();
      const inputs = component.find('input');

      expect(inputs).toHaveLength(3);
    });

    it('should render correctly with labels', () => {
      const component = createComponent();
      const labels = component.find('label');

      expect(labels).toHaveLength(3);
    });

    it('should render correctly with submit button', () => {
      const component = createComponent();
      const submitButton = component.find('button');

      expect(submitButton).toHaveLength(1);
    });
  });

  describe('Actions', () => {
    describe('mapDispatchToProps', () => {
      const dispatch = jest.fn();
      const props = {
        onChange: jest.fn(),
        onCheck: jest.fn(),
        onSubmit: jest.fn(),
      };

      const dispatchResult = mapDispatchToProps(dispatch, props);

      it('should have a `onChange` attribute ', () => expect(dispatchResult).toHaveProperty('onChange'));
      it('should have a `onCheck` attribute ', () => expect(dispatchResult).toHaveProperty('onCheck'));
      it('should have a `onSubmit` attribute ', () => expect(dispatchResult).toHaveProperty('onSubmit'));

      it('should call `onChange` with "test" value', () => {
        dispatchResult.onChange({ target: { value: 'test' } });
        expect(dispatch).toHaveBeenCalledWith(searchBarValue({ target: { value: 'test' } }));
      });

      it('should call `onCheck` with "all" value', () => {
        dispatchResult.onCheck('all');
        expect(dispatch).toHaveBeenCalledWith(searchBarFilter('all'));
      });

      it('should call `onChange` with "test" value', () => {
        dispatchResult.onSubmit();
        expect(dispatch).toHaveBeenCalledWith(searchBarSubmitQuery());
      });
    });
  });
});
