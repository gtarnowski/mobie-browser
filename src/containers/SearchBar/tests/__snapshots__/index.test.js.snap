// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> Render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    filter="all"
    onChange={[MockFunction]}
    onCheck={[MockFunction]}
    onSubmit={[MockFunction]}
    searchValue="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <SearchInput
          onChange={[MockFunction]}
          searchValue="test"
        />
        <div>
          <button
            type="submit"
          >
            Search
          </button>
        </div>
        <div
          className="SearchRadio"
        >
          <input
            checked={true}
            id="All"
            name="filter"
            onChange={[Function]}
            type="radio"
          />
          <label
            htmlFor="All"
            id="All-radio"
          >
            All
          </label>
        </div>
        <div
          className="SearchRadio"
        >
          <input
            checked={false}
            id="Title"
            name="filter"
            onChange={[Function]}
            type="radio"
          />
          <label
            htmlFor="Title"
            id="Title-radio"
          >
            Title
          </label>
        </div>
        <div
          className="SearchRadio"
        >
          <input
            checked={false}
            id="IMDb"
            name="filter"
            onChange={[Function]}
            type="radio"
          />
          <label
            htmlFor="IMDb"
            id="IMDb-radio"
          >
            IMDb
          </label>
        </div>
      </form>,
      "className": "SearchBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchInput
            onChange={[MockFunction]}
            searchValue="test"
          />,
          <div>
            <button
              type="submit"
            >
              Search
            </button>
          </div>,
          Array [
            <div
              className="SearchRadio"
            >
              <input
                checked={true}
                id="All"
                name="filter"
                onChange={[Function]}
                type="radio"
              />
              <label
                htmlFor="All"
                id="All-radio"
              >
                All
              </label>
            </div>,
            <div
              className="SearchRadio"
            >
              <input
                checked={false}
                id="Title"
                name="filter"
                onChange={[Function]}
                type="radio"
              />
              <label
                htmlFor="Title"
                id="Title-radio"
              >
                Title
              </label>
            </div>,
            <div
              className="SearchRadio"
            >
              <input
                checked={false}
                id="IMDb"
                name="filter"
                onChange={[Function]}
                type="radio"
              />
              <label
                htmlFor="IMDb"
                id="IMDb-radio"
              >
                IMDb
              </label>
            </div>,
          ],
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [MockFunction],
            "searchValue": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              type="submit"
            >
              Search
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "All",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                id="All"
                name="filter"
                onChange={[Function]}
                type="radio"
              />,
              <label
                htmlFor="All"
                id="All-radio"
              >
                All
              </label>,
            ],
            "className": "SearchRadio",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "id": "All",
                "name": "filter",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "All",
                "htmlFor": "All",
                "id": "All-radio",
              },
              "ref": null,
              "rendered": "All",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Title",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="Title"
                name="filter"
                onChange={[Function]}
                type="radio"
              />,
              <label
                htmlFor="Title"
                id="Title-radio"
              >
                Title
              </label>,
            ],
            "className": "SearchRadio",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "Title",
                "name": "filter",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Title",
                "htmlFor": "Title",
                "id": "Title-radio",
              },
              "ref": null,
              "rendered": "Title",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "IMDb",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="IMDb"
                name="filter"
                onChange={[Function]}
                type="radio"
              />,
              <label
                htmlFor="IMDb"
                id="IMDb-radio"
              >
                IMDb
              </label>,
            ],
            "className": "SearchRadio",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "IMDb",
                "name": "filter",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "IMDb",
                "htmlFor": "IMDb",
                "id": "IMDb-radio",
              },
              "ref": null,
              "rendered": "IMDb",
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <SearchInput
            onChange={[MockFunction]}
            searchValue="test"
          />
          <div>
            <button
              type="submit"
            >
              Search
            </button>
          </div>
          <div
            className="SearchRadio"
          >
            <input
              checked={true}
              id="All"
              name="filter"
              onChange={[Function]}
              type="radio"
            />
            <label
              htmlFor="All"
              id="All-radio"
            >
              All
            </label>
          </div>
          <div
            className="SearchRadio"
          >
            <input
              checked={false}
              id="Title"
              name="filter"
              onChange={[Function]}
              type="radio"
            />
            <label
              htmlFor="Title"
              id="Title-radio"
            >
              Title
            </label>
          </div>
          <div
            className="SearchRadio"
          >
            <input
              checked={false}
              id="IMDb"
              name="filter"
              onChange={[Function]}
              type="radio"
            />
            <label
              htmlFor="IMDb"
              id="IMDb-radio"
            >
              IMDb
            </label>
          </div>
        </form>,
        "className": "SearchBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SearchInput
              onChange={[MockFunction]}
              searchValue="test"
            />,
            <div>
              <button
                type="submit"
              >
                Search
              </button>
            </div>,
            Array [
              <div
                className="SearchRadio"
              >
                <input
                  checked={true}
                  id="All"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  htmlFor="All"
                  id="All-radio"
                >
                  All
                </label>
              </div>,
              <div
                className="SearchRadio"
              >
                <input
                  checked={false}
                  id="Title"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  htmlFor="Title"
                  id="Title-radio"
                >
                  Title
                </label>
              </div>,
              <div
                className="SearchRadio"
              >
                <input
                  checked={false}
                  id="IMDb"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  htmlFor="IMDb"
                  id="IMDb-radio"
                >
                  IMDb
                </label>
              </div>,
            ],
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChange": [MockFunction],
              "searchValue": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                type="submit"
              >
                Search
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "All",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={true}
                  id="All"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />,
                <label
                  htmlFor="All"
                  id="All-radio"
                >
                  All
                </label>,
              ],
              "className": "SearchRadio",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": true,
                  "id": "All",
                  "name": "filter",
                  "onChange": [Function],
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "All",
                  "htmlFor": "All",
                  "id": "All-radio",
                },
                "ref": null,
                "rendered": "All",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Title",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  id="Title"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />,
                <label
                  htmlFor="Title"
                  id="Title-radio"
                >
                  Title
                </label>,
              ],
              "className": "SearchRadio",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "id": "Title",
                  "name": "filter",
                  "onChange": [Function],
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Title",
                  "htmlFor": "Title",
                  "id": "Title-radio",
                },
                "ref": null,
                "rendered": "Title",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "IMDb",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  id="IMDb"
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                />,
                <label
                  htmlFor="IMDb"
                  id="IMDb-radio"
                >
                  IMDb
                </label>,
              ],
              "className": "SearchRadio",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "id": "IMDb",
                  "name": "filter",
                  "onChange": [Function],
                  "type": "radio",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "IMDb",
                  "htmlFor": "IMDb",
                  "id": "IMDb-radio",
                },
                "ref": null,
                "rendered": "IMDb",
                "type": "label",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
